app/src/main/crypto.ts

```diff
@@ -0,0 +1,332 @@
+      const cmd = ["gpg", "--trust-model", "always"];
+      if (this.gpgHomedir) {
+        cmd.push("--homedir", this.gpgHomedir);
+      }
+      return cmd;
+    }
+  }
+
+  /**
+   * Decrypt a message from encrypted buffer content
+   * Messages are not gzipped, so no decompression is needed (unlike files)
+   * @param encryptedContent - The encrypted message content
+   * @returns Promise<string> - The decrypted plaintext
+   */
+  async decryptMessage(encryptedContent: Buffer): Promise<string> {
+    const cmd = this.getGpgCommand();
+    cmd.push("--decrypt");
+
+    return new Promise((resolve, reject) => {
+      const gpgProcess = spawn(cmd[0], cmd.slice(1), {
+        stdio: ["pipe", "pipe", "pipe"],
+      });
+
+      let stdout = Buffer.alloc(0);
+      let stderr = Buffer.alloc(0);
+
+      // Write encrypted content to GPG stdin
+      gpgProcess.stdin.write(encryptedContent);
+      gpgProcess.stdin.end();
+
+      // Collect stdout (decrypted content)
+      gpgProcess.stdout.on("data", (chunk) => {
+        stdout = Buffer.concat([stdout, chunk]);
+      });
+
+      // Collect stderr (error messages)
+      gpgProcess.stderr.on("data", (chunk) => {
+        stderr = Buffer.concat([stderr, chunk]);
+      });
+
+      gpgProcess.on("close", async (code) => {
+        if (code !== 0) {
+          const errorMessage = stderr.toString("utf8");
+          reject(
+            new CryptoError(
+              `GPG decryption failed (exit code ${code}): ${errorMessage}`,
+            ),
+          );
+          return;
+        }
+
+        // Messages are not gzipped, so return the decrypted content directly as string
+        resolve(stdout.toString("utf8"));
+      });
+
+      gpgProcess.on("error", (error) => {
+        reject(
+          new CryptoError(
+            `Failed to start GPG process: ${error.message}`,
+            error,
+          ),
+        );
+      });
+    });
+  }
+
+  /**
+   * Decrypt a file and return the path to the decrypted file with original filename
+   * Uses streaming approach to handle large files efficiently (similar to legacy client)
+   * @param filepath - Path to the encrypted file
+   * @returns Promise with path to decrypted file and original filename
+   */
+  async decryptFile(
+    filepath: string,
+  ): Promise<{ filePath: string; filename: string }> {
+    const cmd = this.getGpgCommand();
+    cmd.push("--decrypt", filepath);
+
+    return new Promise((resolve, reject) => {
+      // Create temporary directory for this operation
+      const tempDir = fs.mkdtempSync(
+        path.join(os.tmpdir(), "securedrop-decrypt-"),
+      );
+      const tempGpgOutput = path.join(
+        tempDir,
+        `securedrop-gpg-${Date.now()}-${Math.random().toString(36).slice(2)}`,
+      );
+
+      const gpgOutputFile = fs.createWriteStream(tempGpgOutput);
+      let stderr = Buffer.alloc(0);
+
+      const gpgProcess = spawn(cmd[0], cmd.slice(1), {
+        stdio: ["pipe", "pipe", "pipe"],
```

Comment:
from the docs, I think this setting is the default (+ we don't explicitly set it in other places): https://nodejs.org/api/child_process.html#optionsstdio 

if we do want to set explicitly / not trust the default, I'd probably advocate for doing so everywhere we call `child_process.spawn` just for consistency